config/ompi_setup_prrte.m4

```diff
@@ -276,18 +266,21 @@ AC_DEFUN([_OMPI_SETUP_PRRTE_EXTERNAL], [
 
     CPPFLAGS="$opal_prrte_CPPFLAGS_save"
 
+    # If an external build and the user told us where to find PRRTE,
+    # find prterun and save that path.
+    prterun_path=
     AS_IF([test "$setup_prrte_external_happy" = "yes"],
-          [AS_IF([test -n "$with_prrte"],
-                 [PRTE_PATH="${with_prrte}/bin/prte"
-                  AS_IF([test ! -r ${PRTE_PATH}],
-                        [AC_MSG_ERROR([Could not find prte binary at $PRTE_PATH])],
-                        [PRTE_PATH="${with_prrte}/bin"])],
-                 [PRTE_PATH=""
-                  OPAL_WHICH([prte], [PRTE_PATH])
-                  AS_IF([tets -z "$PRTE_PATH"],
-                        [AC_MSG_WARN([Could not find prte in PATH])
-                         setup_prrte_external_happy=no],
-                        [PRTE_PATH="`echo $PRTE_PATH | sed -e 's/\/prte//'`"])])])
+          [AS_IF([test "${with_prrte_bindir}" = "yes" -o "${with_prrte_bindir}" = "no"],
```

Comment:
Either one should be fine.  The version here expands the input variables for both cases even if the first is true.  The || notation can end up spawning extra subshells.  In this case, we don't care about the early expansion (since it's the same variable in both cases), so either works and I lean towards the one that is shorter and doesn't cause extra subshells.

Reply:
I thought I had a dim recollection about portability issues with `-o` and `-a` (perhaps BSD `test` doesn't -- or didn't, at some point in the past? -- support those operators).  Seems to work on MacOS 12.3 `/bin/test`, so let's just forget I asked.  😄